R/iterators.R

```diff
@@ -306,7 +317,7 @@ create_es <- function(graph, idx, na_ok = FALSE) {
 
 simple_vs_index <- function(x, i, na_ok = FALSE) {
   res <- unclass(x)[i]
-  if (!na_ok && any(is.na(res))) stop('Unknown vertex selected')
+  if (!(na_ok | !any(is.na(res)))) stop('Unknown vertex selected')
```

Comment:
I don't exactly understand this change. In your commit message you say

> This prevents `any(is.na(vs))` from getting evaluated if `na.ok = TRUE`

but isn't this also the case with the original conditional `!na_ok && any(is.na(res))`? That is, if `na_ok = TRUE`, there is no need to evaluate the remaining condition, since the entire condition is already known to evaluate to `FALSE`. Or is there some specific reason why this is not the case here?

Reply:
Not sure about this; there would clearly be a difference here if `na_ok` is a vector because the `&&` operator is not vectorized (it only looks at the first argument of `na_ok` if it is a vector), while `|` is vectorized. However, I think `na_ok` is meant to be a single logical value anyway so I don't think this change is necessary.